### ISSUE TITLE
I can not connect to the MQTT broker via a TLS connection if DHCP is not used

### ISSUE THREAD

**Issue by ashendrik** [esp8266/Arduino#5826]
----------------------------- Delete below -----------------------------

If your issue is a general question, starts similar to "How do I..", is related to 3rd party libs, or is related to hardware, please discuss at a community forum like esp8266.com.

INSTRUCTIONS
============
If you do not follow these instructions, your issue may be dismissed.

1. Follow the checklist under Basic Infos and fill in the [ ] spaces with an X.
2. Fill in all the fields under Platform and Settings in IDE marked with [ ] (pick the correct option for you in each case, delete the others).
3. If you haven't already done so, test your issue against current master branch (aka latest git), because it may have been already fixed.
4. Describe your problem.
5. If you have a STACK DUMP decode it:

https://arduino-esp8266.readthedocs.io/en/latest/Troubleshooting/stack_dump.html

6. Include a Minimal Complete Reproducible Example sketch that shows your issue. Do not include your entire project, or a huge piece of code.
7. Include debug messages:

https://arduino-esp8266.readthedocs.io/en/latest/Troubleshooting/debugging.html 

8. Use markup (buttons above) and the Preview tab to check what the issue will look like.
9. Delete these instructions from the above to the below marker lines before submitting this issue.

----------------------------- Delete above -----------------------------

### Basic Infos

- [+] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [+] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [+] I have tested that the issue is present in current master branch (aka latest git).
- [+] I have searched the issue tracker for a similar issue.
- [+] If there is a stack dump, I have decoded it.
- [+] I have filled out all fields below.

#### Platform

- Hardware:				[ESP-12]
- Core Version:			[2.5.0]
- Development Env:		[Arduino IDE]
- Operating System:		[MacOS]

### Settings in IDE

- Module:  			[Nodemcu]
- Flash Mode:			[dio]
- Flash Size: 	         	[4MB]
- lwip Variant:                  [v2 Lower Memory]
- Reset Method:		[nodemcu]
- Flash Frequency:	        [40Mhz]
- CPU Frequency:		[80Mhz]
- Upload Using:		[SERIAL]
- Upload Speed:              [115200] (serial upload only)

### Problem Description

I can not connect to the MQTT broker via a TLS connection if DHCP is not used.
In other words, if I add one line "WiFi.config (localIP, gateWay, subNet, DNS1, DNS2 )" to the code, I can never connect to the MQTT broker. PubSubClient state() return -2, the network connection failed.
If this line is not present, the connection is almost always successful at the first attempt.

**Comment by devyte** [comment]
No MCVE, and issue is related to 3rd party lib that isn't hosted in this repo.
I suggest you ask the author of the mqtt lib and try to reduce the issue to a simple WiFiClientSecure connection attempt. If you can reduce it to that, please open a new issue here, fill out the requested info in the issue template, and provide the code of the reduced MCVE.
Closing.
